transports/http/auth/require_auth.go

```diff
@@ -14,19 +14,27 @@ import (
 func RequireAdmin(handler gin.HandlerFunc, requireAdmin bool) gin.HandlerFunc {
 	if requireAdmin {
 		return func(c *gin.Context) {
-			token, exist := c.Get("token")
-			if !exist {
-				c.AbortWithStatus(http.StatusUnauthorized)
+			if err := _validateToken(c); err == nil {
+				handler(c)
+			} else {
+				c.AbortWithStatusJSON(http.StatusUnauthorized, err.Error())
 			}
-			t, ok := token.(*domains.Token)
-			if !ok {
-				c.AbortWithStatusJSON(http.StatusUnauthorized, errors.New("something went wrong"))
-			}
-			if !t.IsAdmin {
-				c.AbortWithStatus(http.StatusUnauthorized)
-			}
-			handler(c)
 		}
 	}
 	return handler
 }
+
+func _validateToken(c *gin.Context) error {
```

Comment:
why this `_` prefix?

Reply:
I thought - based on our code base - that we have such convention, that file-local functions have underscore prefix but if I was wrong, I'm ok to rename it to `validateToken`.

Reply:
this convention is only in buxorg, because almost everything there was in one packege. Here we tried to have everything in it's package, so please remove `_` 🙂